### PR TITLE
feat(gatsby-core-utils): Add file download functions

### PR DESCRIPTION
**Pull request by ascorbic**
Move the file download function `fetchRemoteNode` to gatsby-core-utils and rename it to `fetchRemoteFile`

### CODE REVIEW

**Review by laurieontech — COMMENTED** [review]
Does this solve the issue entirely though? We use both remoteFileNode and createFileNode. This appears to only move the former out of gatsby-source-filesystem?

**Comment by ascorbic** [comment]
The idea would be that we stop creating file nodes, and just download the file

File: packages/gatsby-core-utils/src/fetch-remote-file.ts [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+        if (timeout) {
+          clearTimeout(timeout)
+        }
+        resolve(response)
+      })
+    })
+  })
+
+interface IFetchRemoteFileOptions {
```
Comment (wardpeet) [inline packages/gatsby-core-utils/src/fetch-remote-file.ts:146]: can we move this to the top of the file so types are in a single place?

**Review by wardpeet — COMMENTED** [review]
This looks good! This is going to be helpful for other plugins as well. Can you port some tests over?

**Review by laurieontech — APPROVED** [review]
Yay tests!

**Pull request merged**